shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/data/repository/SurveyRepositoryImpl.kt

```diff
@@ -0,0 +1,41 @@
+package co.nimblehq.ic.kmm.suv.data.repository
+
+import co.nimblehq.ic.kmm.suv.data.local.datasource.SurveyLocalDataSource
+import co.nimblehq.ic.kmm.suv.data.local.model.toSurvey
+import co.nimblehq.ic.kmm.suv.data.remote.datasource.SurveyRemoteDataSource
+import co.nimblehq.ic.kmm.suv.data.remote.model.toSurvey
+import co.nimblehq.ic.kmm.suv.data.remote.model.toSurveyRealmObject
+import co.nimblehq.ic.kmm.suv.data.remote.parameter.GetSurveysApiQueryParams
+import co.nimblehq.ic.kmm.suv.domain.model.Survey
+import co.nimblehq.ic.kmm.suv.domain.repository.SurveyRepository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.last
+
+class SurveyRepositoryImpl(
+    private val surveyRemoteDataSource: SurveyRemoteDataSource,
+    private val surveyLocalDataSource: SurveyLocalDataSource
+) : SurveyRepository {
+
+    override fun getSurveys(pageNumber: Int, pageSize: Int): Flow<List<Survey>> {
+        // TODO: Check this logic again when we have pull to refresh
+        return flow {
+            val localSurveys = surveyLocalDataSource.getSurveys().map { it.toSurvey() }
+            if (localSurveys.isNotEmpty()) {
+                emit(localSurveys)
+            }
+
+            val apiSurveys = surveyRemoteDataSource
+                .getSurveys(GetSurveysApiQueryParams(pageNumber, pageSize))
+                .last()
+
+            val latestSurveys = apiSurveys.map { it.toSurvey() }
+
+            if (localSurveys != latestSurveys) {
+                emit(latestSurveys)
+            }
+
+            surveyLocalDataSource.saveSurveys(apiSurveys.map { it.toSurveyRealmObject() })
```

Comment:
Should we clear the current data list before copying new list to Realm?

Reply:
I prefer to keep the current data list in the Realm database, for example, if we load more data (page 2), it's better to keep the data of the page 1.

Reply:
I improved the logic a little bit here: https://github.com/suho/kmm-ic/pull/102/commits/90445f1173cd31c1ed11874b276208ea228c3957

What do you think?

Reply:
@suho lgtm, we will need it for pull-down to refresh the Home screen data feature. Or when loading the the Home screen at the first page.